### PR TITLE
ansible: refact trss

### PR DESCRIPTION
**Pull request by zdtsw**
- use unarchive download and unpack nodejs
	- use apt_* module handle mongodb key repo
	- use user module create jenkins
	- use blockinfile append two lines in nginx default config
	- move tasks to to more consistent

##### Checklist

- [x] commit message has one of the [standard prefixes](https://github.com/adoptium/infrastructure/blob/master/CONTRIBUTING.md#commit-messages)
- [ ] [faq.md](https://github.com/adoptium/infrastructure/blob/master/FAQ.md) updated if appropriate
- [ ] other documentation is changed or added (if applicable)
- [ ] playbook changes run through [VPC](https://ci.adoptopenjdk.net/view/Tooling/job/VagrantPlaybookCheck/) or [QPC](https://ci.adoptopenjdk.net/view/Tooling/job/QEMUPlaybookCheck/) (if you have access)
- [ ] for inventory.yml changes, bastillion/nagios/jenkins updated accordingly

### CODE REVIEW

File: ansible/playbooks/AdoptOpenJDK_Unix_Playbook/trss.yml [hunk 1]
```diff
@@ -23,18 +23,12 @@
...
-        - name: Download Nodev12.16.1 tarball
-          get_url:
-            url: https://nodejs.org/dist/v12.16.1/node-v12.16.1-linux-x64.tar.xz
-            dest: /tmp/node-v12.16.1-linux-x64.tar.xz
-            mode: 0440
-          tags: node_install
-
-        - name: Extract Node tarball
+        # Prepare for node.js
+        - name: Download Node v12.16.1 tarball and extract Node tarball
```
Comment (karianna) [inline ansible/playbooks/AdoptOpenJDK_Unix_Playbook/trss.yml:27]: Do we need a newer node version?

**Review by karianna — CHANGES_REQUESTED** [review]

File: ansible/playbooks/AdoptOpenJDK_Unix_Playbook/trss.yml (hunk 1 above)
Comment (zdtsw) [reply ansible/playbooks/AdoptOpenJDK_Unix_Playbook/trss.yml:27]: we can, but i would prefer having another PR to upgrade node version.
at least need to test with trss first

**Review by sxa — COMMENTED** [review]
Just one question. Also inviting Haroon for review.

File: ansible/playbooks/AdoptOpenJDK_Unix_Playbook/trss.yml [hunk 2]
```diff
@@ -70,32 +67,38 @@
...
             state: started
           tags: mongodb
 
         - name: Create Jenkins user
-          action: user name={{ Jenkins_Username }} state=present shell=/bin/bash
+          user:
+            name: "{{ Jenkins_Username }}"
+            shell: /bin/bash
+            stat: present
+            uid: 1040
```
[the remark this answers is not in the document]
Comment (zdtsw) [reply ansible/playbooks/AdoptOpenJDK_Unix_Playbook/trss.yml:94]: good spot!
it is not intended to be there at all, i will remove it

File: ansible/playbooks/AdoptOpenJDK_Unix_Playbook/trss.yml (hunk 1 above)
Comment (zdtsw) [reply ansible/playbooks/AdoptOpenJDK_Unix_Playbook/trss.yml:27]: sure, will create a new issue for upgrade

Comment (zdtsw) [reply ansible/playbooks/AdoptOpenJDK_Unix_Playbook/trss.yml:27]: FYI: https://github.com/adoptium/infrastructure/issues/2591

Comment (sxa) [reply ansible/playbooks/AdoptOpenJDK_Unix_Playbook/trss.yml:27]: @karianna Are you ok for this to go in now and the node update evaluated in a separate PR. We'll be doing some changes relating to how TRSS is deployed in the next few weeks so I suspect we'll look at the Node upgrade alongside that work.

Comment (zdtsw) [reply ansible/playbooks/AdoptOpenJDK_Unix_Playbook/trss.yml:27]: @karianna can you have another review on this PR, thanks!

**Comment by smlambert** [comment]
Adding @llxia for awareness

**Comment by llxia** [comment]
Currently, from what I can see, TRSS still runs from this dir - `/home/jenkins/...`
Nothing needs to be changed for now. But since we are going to update TRSS to run under `/data`, playbook needs to be updated at a later stage.

File: ansible/playbooks/AdoptOpenJDK_Unix_Playbook/trss.yml [hunk 3]
```diff
@@ -128,28 +131,21 @@
             state: present
           tags: nginx_conf
 
-        - name: Add server name trss.adoptopenjdk.net /etc/nginx/sites-enabled/default
+        - name: Replace server name with trss.adoptopenjdk.net /etc/nginx/sites-enabled/default
```
[the remark this answers is not in the document]
Comment (zdtsw) [reply ansible/playbooks/AdoptOpenJDK_Unix_Playbook/trss.yml:134]: I've been thinking, the redirect is probably done in ELB, if TRSS is currently running in AWS
if we change code in ansible to replace "trss.adoptopenjdk.net" with "trss.adoptium.net" could proportionally be two side effects:
- Loadbalancer might need an update.
- the code wont find line "`server_name _`" in the current nginx config (which has been modified as `"server_name trss.adoptopenjdk.net"` so it wont really do any change.

**Comment by sxa** [comment]
> Currently, from what I can see, TRSS still runs from this dir - `/home/jenkins/...` Nothing needs to be changed for now. But since we are going to update TRSS to run under `/data`, related code in the playbook needs to be updated at a later stage.

Technically it's running from `/data` now already - `openjdk-test-tools` and `aqa-test-tools` in the user's homedir are symlinked into `/data`

**Comment by llxia** [comment]
Is symlink a temporary solution? If it is a temporary solution, then we need to update all references to `/data`. If not, we need to update the playbook with the symlink.

**Comment by zdtsw** [comment]
Can I get some input for this PR, if anything else is unclear or require change? thanks!
@llxia @sxa @karianna

**Review by karianna — APPROVED** [review]
LGTM

**Pull request merged**